.github/workflows/pr-build-image.yaml

```
@@ -92,25 +92,8 @@ jobs:
       - name: Get the last commit short SHA of the PR
         uses: ./.github/actions/get-sha
 
-      - name: Check if Image Already Exists
-        if: env.proceed_with_build == 'true'
-        run: |
-          IMAGE_TAG="pr-${{ github.event.number }}"
-          IMAGE_NAME="${{ env.REGISTRY }}/rhdh-community/rhdh:${IMAGE_TAG}"
-
-          # Check if any image tag exists for the PR
-          IMAGE_TAGS=$(curl -s "https://quay.io/api/v1/repository/rhdh-community/rhdh/tag/" | jq -r --arg tag "$IMAGE_TAG" '.tags[] | select(.name | startswith($tag)) | .name')
-
-          if [ -n "$IMAGE_TAGS" ]; then
-            echo "Image $IMAGE_NAME or its variants already exist. Skipping the build."
-            echo "image_exists=true" >> $GITHUB_ENV
-          else
-            echo "Image $IMAGE_NAME does not exist. Proceeding with the build."
-            echo "image_exists=false" >> $GITHUB_ENV
-          fi
-
       - name: Build and Push with Buildx
-        if: env.proceed_with_build == 'true' && env.image_exists == 'false'
+        if: env.relevant_changes == 'true' || env.image_exists == 'false'
```

```suggestion
        if: env.relevant_changes == 'true' || env.image_exists == 'false'
```
Shouldn't this be the same as before? Wouldn't we want a new image only when we have relevant changes and the image corresponding to these changes doesn't exist?

---

The current logic ensures that a new image is only built when there are relevant changes (proceed_with_build == 'true') and an image corresponding to these changes does not already exist (image_exists == 'false). This avoids unnecessary builds while still ensuring that new changes result in a fresh image. Altering this logic would either trigger unnecessary builds or skip builds when they are actually required.

---

Yeah, so shouldn't we still be using && instead of || ?